src/commands/deploy/showCoreToolsWarning.ts

```diff
@@ -0,0 +1,61 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { DialogResponses, IActionContext, nonNullValue, openUrl } from "@microsoft/vscode-azext-utils";
+import { MessageItem } from "vscode";
+import { getLocalFuncCoreToolsVersion } from "../../funcCoreTools/getLocalFuncCoreToolsVersion";
+import { FuncVersion, funcVersionLink, tryParseFuncVersion } from "../../FuncVersion";
+import { localize } from "../../localize";
+import { getWorkspaceSetting, updateGlobalSetting } from "../../vsCodeConfig/settings";
+
+export async function showCoreToolsWarning(context: IActionContext, projectVersion: FuncVersion): Promise<void> {
+    const showCoreToolsWarningKey: string = 'showCoreToolsWarning';
+    if (getWorkspaceSetting<boolean>(showCoreToolsWarningKey)) {
+        const coreToolsVersion = await getLocalFuncCoreToolsVersion(context, undefined)
+        if (coreToolsVersion !== null) {
+            const localVersion = nonNullValue(tryParseFuncVersion(coreToolsVersion), 'localCoreToolsVersion');
+            let result: MessageItem | undefined;
+
+            if (localVersion === FuncVersion.v2 || localVersion === FuncVersion.v3) {
+                result = await showCoreToolsEOLWarning(context, localVersion);
+                while (result === DialogResponses.learnMore) {
+                    await openUrl(funcVersionLink);
+                    result = await showCoreToolsEOLWarning(context, localVersion);
+                }
+                if (result === DialogResponses.dontWarnAgain) {
```

Comment:
I believe that you can consolidate this with L32. If you move it out of the conditional and then just check the result at the end (as in on L40), you should only have to check once.